demo/cca/manual-end-to-end.md

```diff
@@ -0,0 +1,295 @@
+
+
+* For build to succeed one needs to install following packages:
+1. protoc-gen-go with version v1.26
+2. protoc-gen-go-grpc version v1.1
+3. protoc-gen-go-json version v1.1.0
+4. mockgen version v1.6.0
+
+* Commands below assume execution in a Bourne-compatible shell. Please adjust appropriately in case any other shell is used.
+
+
+## Creation of CCA Endorsements
+
+First create new Concise Module Identifiers (CoMID's) and use them in creating Concise Reference Integrity Manifests (CoRIM's) using reference templates located under `docs/demo/cca/prov-verif-e2e/data/templates` to provision them in Veraison Verification Service.
+
+More details about CoRIM and CoMID can be found [here](https://datatracker.ietf.org/doc/draft-ietf-rats-corim/)
+
+### Initial Setup
+
+Do this ONLY if it is not done already:
+PLEASE NOTE: Setup instructions are same between PSA and CCA demonstration.
+
+In a new bourne shell session
+
+```shell
+export TOPDIR=$(pwd)
+```
+
+* Install `cocli` tool for provisioning endorsements using following command
+
+```shell
+go install github.com/veraison/corim/cocli@demo-cca-1.0.0
+```
+
+* Install `arc` tool for decoding attestation results using the following command
+
+```shell
+go install github.com/veraison/ear/arc@demo-cca-1.0.0
+```
+
+```shell
+git clone https://github.com/veraison/docs
+```
+
+* Remember this shell as shell-1. 
+
+### Create CoMID's
+
+```shell
+cd ${TOPDIR}/docs/demo/cca/prov-verif-e2e
+```
+
+* Create CoMIDs for CCA Trust Anchors and Reference Values using given JSON template
+
+Please inspect template JSON file `data/templates/comid-cca-ta-endorsements.json`  to provision cca trust anchors (ta) and modify anything as per your requirement
+
+Please inspect template JSON file `data/templates/comid-cca-mult-refval.json` to provision cca reference values and modify anything as per your requirement
+
+Using single command one can create two CBOR files for two CoMIDs from above json templates
+
+```shell
+cocli comid create --template=data/templates/comid-cca-ta-endorsements.json \
+                    --template=data/templates/comid-cca-mult-refval.json
+```
+
+One should see, on the console
+created "comid-cca-ta-endorsements.cbor" from "data/templates/comid-cca-ta-endorsements.json"
+created "comid-cca-mult-refval.cbor" from "data/templates/comid-cca-mult-refval.json"
+
+### Create CoRIM
+
+* Create a single CoRIM using given CoRIM JSON template and supplying CBOR Trust Anchor and Reference Value CoMIDs from above
+
+```shell
+cocli corim create --template=data/templates/corim-cca.json --comid=comid-cca-ta-endorsements.cbor --comid=comid-cca-mult-refval.cbor
+```
+created "corim-cca.cbor" from "data/templates/corim-cca.json"
+
+### Using newly generated CoRIM
+Move the generated CORIM above to data/cbor directory
+ 
+```shell
+mv corim-cca.cbor  data/cbor/
+```
+* Please retain this shell, as shell-1 as you would need to come back here at the time of provisioning the endorsements.
```

Comment:
There is already a shell that we've been asked to rememeber as "shell-1"  on line 108, that has not been "dereferenced" up to this point. 

Perhaps giving a more descriptive name than "shell-1" wold be usefull here, e.g. the "client shell".

Reply:
Clarified the usage of shell-1